packages/lib/CalendarService.ts

```diff
@@ -241,25 +243,64 @@ export default abstract class BaseCalendarService implements Calendar {
       )
     ).flat();
 
-    const events = objects
-      .filter((e) => !!e.data)
-      .map((object) => {
-        const jcalData = ICAL.parse(object.data);
-        const vcalendar = new ICAL.Component(jcalData);
-        const vevent = vcalendar.getFirstSubcomponent("vevent");
-        const event = new ICAL.Event(vevent);
-        const vtimezone = vcalendar.getFirstSubcomponent("vtimezone");
-        if (vtimezone) {
-          const zone = new ICAL.Timezone(vtimezone);
-          event.startDate = event.startDate.convertToZone(zone);
-          event.endDate = event.endDate.convertToZone(zone);
+    const events: { start: string; end: string }[] = [];
+
+    objects.forEach((object) => {
+      if (object.data == null) return;
+
+      const jcalData = ICAL.parse(object.data);
+      const vcalendar = new ICAL.Component(jcalData);
+      const vevent = vcalendar.getFirstSubcomponent("vevent");
+      const event = new ICAL.Event(vevent);
+      const vtimezone = vcalendar.getFirstSubcomponent("vtimezone");
+
+      if (event.isRecurring()) {
+        let maxIterations = 365;
+        if (["HOURLY", "SECONDLY", "MINUTELY"].includes(event.getRecurrenceTypes())) {
+          console.error(`Won't handle [${event.getRecurrenceTypes()}] recurrence`);
+          return;
```

Comment:
I can implement this if you like. 

But consider this: Imagine creating a secondly event last year and anyone want to book on the 31st of December. It will take `31536000` (`365*24*60*60`) iterations worst case. I would rather encourage users not to create these kind of events in the first place 😂 

---

Is there any way to see how often this logging will occur?

Reply:
Good enough! We can check the Logs in Datadog for this occurrence.